src/ServiceControl/Recoverability/Archiving/ArchiveOperationLogic.cs

```diff
@@ -0,0 +1,128 @@
+﻿namespace ServiceControl.Recoverability
+{
+    using ServiceControl.Infrastructure.DomainEvents;
+    using System;
+
+    public class ArchiveOperationLogic
```

Comment:
Isn't this just a ArchiveOperation? "Logic" is one of those fuzzy words that don't add anything concrete to me.